include/alpaka/kernel/Traits.hpp

```diff
@@ -202,6 +202,29 @@ namespace alpaka
         "-Wdocumentation" // clang does not support the syntax for variadic template arguments "args,..."
 #endif
 
+
+    /// Check if a type used as kernel argument is trivially copyable
+    ///
+    /// \attention In case this trait is specialized for a user type the user should be sure that the result of calling
+    /// the copy constructor is equal to use memcopy to duplicate the object. An existing destructor should be side
+    /// effect free.
+    ///
+    /// It's implementation defined whether the closure type of a lambda is trivially copyable.
+    /// Therefor the default implementation is true for trivially copyable or empty (stateless) types.
+    ///
+    /// @tparam T type to check
+    /// @{
+    template<typename T>
```

Comment:
Adding an additional template parameter `typename = void` makes specialization usually simpler.

```suggestion
    template<typename T, typename = void>
```

Instead of creating a specialization for every type, it can be generalized like:
```
template<typename Real>
struct IsKernelArgumentTriviallyCopyable<Real, typename std::enable_if<codi::isForwardType<Real>::value>::type
```

Reply:
@MaxSagebaum thanks for the suggestion, I think it makes sense

Reply:
You guys are going to love `requires` clauses in C++20.

Reply:
good point I changed the template signature